### PR TITLE
In-memory engine: fix online configuration not taking effect for some configs

### LINKED ISSUE #17578: In-memory engine: online configuration change does not take effect for some configs
## Bug Report

```RUST
pub fn new(
      core: Arc<RangeCacheMemoryEngineCore>,
      pd_client: Arc<dyn PdClient>,
      config: Arc<VersionTrack<RangeCacheEngineConfig>>,
      memory_controller: Arc<MemoryController>,
      region_info_provider: Option<Arc<dyn RegionInfoProvider>>,
  ) -> Self {
    let worker = Worker::new("ime-bg");
    let (runner, delete_range_scheduler) = BackgroundRunner::new(
        core.clone(),
        memory_controller,
        region_info_provider.clone(),
        &config,
        pd_client.clone(),
    );
    let config = config.value();
    let scheduler = worker.start_with_timer("ime-bg-runner", runner);

    let (ticker, tx) = BgWorkManager::start_tick(
        scheduler.clone(),
        pd_client,
        config.gc_interval.0,
        config.load_evict_interval.0,
    );

    Self {
        worker,
        scheduler,
        delete_region_scheduler: delete_range_scheduler,
        tick_stopper: Some((ticker, tx)),
        core,
        region_info_provider,
    }
}
```

The config value is used as raw value, so the config changes cannot be perceived.

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Close #17578 Ref #16141

What's Changed:

```commit-message
fix online configuration not taking effect for some configs
```
1. Fix using raw value for configs that support online configuration change.
2. Add validation for config change.

image

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by overvenus — APPROVED** [review]

File: components/server/src/common.rs [hunk 1]
```diff
@@ -76,6 +76,7 @@ const DEFAULT_QUOTA_LIMITER_TUNE_INTERVAL: Duration = Duration::from_secs(5);
 // some services into `to_stop`.
 pub struct TikvServerCore {
     pub config: TikvConfig,
+    pub ime_config: Arc<VersionTrack<RangeCacheEngineConfig>>,
```
[the remark this answers is not in the document]
Comment (SpadeA-Tang) [reply components/server/src/common.rs:79]: If not, the config will be passed by parameter and return value which is not neat.

[the remark this answers is not in the document]
Comment (glorv) [reply components/server/src/common.rs:79]: But this is not neat either. /cc @Connor1996 @LykxSassinator any ideas?

[the remark this answers is not in the document]
Comment (LykxSassinator) [reply components/server/src/common.rs:79]: Agree. It's not appropriate to make `ime_config` as an member of `TikvServerCore` and be exposed to developer, and it should be a member of `TikvConfig`.
IMO, the previous implementation is better, using `VersionTrack::new(range_cache_engine_config)`.

[the remark this answers is not in the document]
Comment (SpadeA-Tang) [reply components/server/src/common.rs:79]: The previous implementation is not worked.
`Arc<VersionTrack<RangeCacheEngineConfig>>` is used both in
`TikvServer::<CER, F>::init(config, service_event_tx.clone());` and
`tikv.init_raw_engines(listener);`
So either saved it in TikvServerCore or returned by `init` and pass to `init_raw_engines` as a parameter.

[the remark this answers is not in the document]
Comment (LykxSassinator) [reply components/server/src/common.rs:79]: then how about change the definition of the callback by using `Option<Box<xxx>>` and add a new function for `RegionInfoAccessor` to support update / replace it ?

And by this way, u can delay the binding of callback in `RegionInfoAccessor` by doing `replace(Some(Box::new(callback)))` when `tikv.init_server(...)`, and there is just one single `Arc<VersionTrack<RangeCacheEngineConfig>>` to be used.

[the remark this answers is not in the document]
Comment (SpadeA-Tang) [reply components/server/src/common.rs:79]: image
If changed cb to Option, then the scheduler cannot be created or we need a task to help RegionCollector to set the cb which is not neat also.
Or we can make RegionInfoAccessor an Option `TikvServer` and I have checked there's no usage of it before `init_raw_engines`.

**Review by LykxSassinator — APPROVED** [review]

**Pull request merged**